core/src/main/java/org/apache/druid/timeline/partition/ShardSpec.java

```diff
@@ -125,8 +125,13 @@ default short getAtomicUpdateGroupSize()
   boolean possibleInDomain(Map<String, RangeSet<String>> domain);
 
   /**
-   * Returns true if two segments of this and other shardSpecs can exist in the same time chunk.
+   * Returns true if this shardSpec and the given {@link PartialShardSpec} share the same partition space.
+   * Any implementation of {@link OverwriteShardSpec} should return true.
+   *
+   * @see PartitionIds
    */
-  @JsonIgnore
-  boolean isCompatible(Class<? extends ShardSpec> other);
+  default boolean sharePartitionSpace(PartialShardSpec partialShardSpec)
+  {
+    return !partialShardSpec.useNonRootGenerationPartitionSpace();
```

Comment:
Should this be `partialShardSpec.useNonRootGenerationPartitionSpace();` instead of `!partialShardSpec.useNonRootGenerationPartitionSpace();` ?

Reply:
Seems like the java doc for ShardSpec#sharePartitionSpace and OverwriteShardSpec#sharePartitionSpace should be updated

Reply:
Ah, all `ShardSpec`s share the same root-generation partition space except `OverwriteShardSpec`. So, the current code is correct. I updated the javadoc to make it more clear.